src/library-authoring/components/ContainerDeleter.tsx

```diff
@@ -30,24 +33,92 @@ const ContainerDeleter = ({
   const restoreContainerMutation = useRestoreContainer(containerId);
   const { showToast } = useContext(ToastContext);
+  const { hits } = useContentFromSearchIndex([containerId]);
+  const containerData = (hits as ContainerHit[])?.[0];
+  const {
+    data: dataDownstreamLinks,
+  } = useContainerEntityLinks({ upstreamContainerKey: containerId });
+  const downstreamCount = dataDownstreamLinks?.length ?? 0;
+
+  const messageMap = useMemo(() => {
+    const containerType = containerData?.blockType;
+    let parentCount = 0;
+    let parentMessage: React.ReactNode;
+    switch (containerType) {
+      case ContainerType.Section:
+        return {
+          title: intl.formatMessage(messages.deleteSectionWarningTitle),
+          parentMessage: '',
+          // Update below fields when sections are linked to courses
+          courseCount: downstreamCount,
+          courseMessage: messages.deleteSectionCourseMessaage,
+        };
+      case ContainerType.Subsection:
+        parentCount = containerData?.sections?.displayName?.length || 0;
+        if (parentCount === 1) {
+          parentMessage = intl.formatMessage(
+            messages.deleteSubsectionParentMessage,
+            { parentName: <b>{containerData?.sections?.displayName?.[0]}</b> },
+          );
+        } else if (parentCount > 1) {
+          parentMessage = intl.formatMessage(messages.deleteSubsectionMultipleParentMessage, {
+            parentCount: <b>{parentCount}</b>,
+          });
+        }
+        return {
+          title: intl.formatMessage(messages.deleteSubsectionWarningTitle),
+          parentMessage,
+          // Update below fields when subsections are linked to courses
+          courseCount: downstreamCount,
+          courseMessage: messages.deleteSubsectionCourseMessaage,
+        };
+      default:
+        parentCount = containerData?.subsections?.displayName?.length || 0;
+        if (parentCount === 1) {
+          parentMessage = intl.formatMessage(
+            messages.deleteUnitParentMessage,
+            { parentName: <b>{containerData?.subsections?.displayName?.[0]}</b> },
+          );
+        } else if (parentCount > 1) {
+          parentMessage = intl.formatMessage(messages.deleteUnitMultipleParentMessage, {
+            parentCount: <b>{parentCount}</b>,
+          });
+        }
+        return {
+          title: intl.formatMessage(messages.deleteUnitWarningTitle),
+          parentMessage,
+          // Update below fields when unit are linked to courses
+          courseCount: downstreamCount,
+          courseMessage: messages.deleteUnitCourseMessage,
+        };
+    }
+  }, [containerData, downstreamCount, messages, intl]);
 
-  // TODO: support other container types besides 'unit'
-  const deleteWarningTitle = intl.formatMessage(messages.deleteUnitWarningTitle);
   const deleteText = intl.formatMessage(messages.deleteUnitConfirm, {
     unitName: <b>{displayName}</b>,
```

Comment:
Since we're now [fetching the `containerData` from the search index](https://github.com/openedx/frontend-app-authoring/pull/2145/files#diff-05afe127c00f49cfcf7264f751492a43d5b12c68da8ac0270ec55aacde79927fR36-R37), we don't need to pass in the `displayName` anymore; we can just use the `containerData.displayName`.

Reply:
Had to fix a minor bug with invalidating single search index queries for this to work properly on name change.